bpy_speckle/connector/ui/account_selection_dialog.py

```diff
@@ -0,0 +1,143 @@
+    bl_label = "Select Account"
+    bl_description = "Select account"
+
+    account_index: bpy.props.IntProperty(default=0)  # type: ignore
+
+    def invoke(self, context: Context, event: Event) -> set[str]:
+        wm = context.window_manager
+        # Clear existing accounts
+        wm.speckle_accounts.clear()
+
+        # Save selected account
+        current_account_index = 0
+
+        # Fetch accounts
+        for i, (id, user_name, server_url, user_email) in enumerate(
+            get_account_enum_items()
+        ):
+            account: speckle_account = wm.speckle_accounts.add()
+            account.id = id
+            account.user_name = user_name
+            account.server_url = server_url
+            account.user_email = user_email
+            if id == wm.selected_account_id:
+                current_account_index = i
+
+        self.account_index = current_account_index
+        return context.window_manager.invoke_props_dialog(self)
+
+    def draw(self, context: Context) -> None:
+        layout = self.layout
+        wm = context.window_manager
+        row = layout.row()
+        # add account button
+        if wm.selected_account_id == "NO_ACCOUNTS":
+            add_account_button_text = "Sign In"
+            add_account_button_icon = "WORLD"
+        else:
+            add_account_button_text = "Add Account"
+            add_account_button_icon = "ADD"
+        row.operator(
+            "speckle.add_account",
+            icon=add_account_button_icon,
+            text=add_account_button_text,
+        )
+
+        if wm.selected_account_id != "NO_ACCOUNTS":
+            layout.template_list(
+                "SPECKLE_UL_accounts_list",
+                "",
+                context.window_manager,
+                "speckle_accounts",
+                self,
+                "account_index",
+            )
+
+    def execute(self, context: Context) -> set[str]:
+        wm = context.window_manager
+        # update the selected account id
+        account = get_account_from_id(wm.speckle_accounts[self.account_index].id)
+        wm.selected_account_id = account.id
+        self.report(
+            {"INFO"},
+            f"Selected account: {account.userInfo.name} - {account.userInfo.email} - {account.serverInfo.url}",
+        )
+        update_workspaces_list(context)
+        update_projects_list(context)
+        # redraw the area
+        context.area.tag_redraw()
+        return {"FINISHED"}
+
+
+def update_workspaces_list(context: Context) -> None:
+    wm = context.window_manager
+    wm.speckle_workspaces.clear()
+    workspaces = get_workspaces(wm.selected_account_id)
+    for id, name in workspaces:
+        workspace: speckle_workspace = wm.speckle_workspaces.add()
+        workspace.id = id
+        workspace.name = name
+    wm.selected_workspace_id = get_default_workspace_id(wm.selected_account_id)
+    print("Updated Workspaces List!")
+
+
+def update_projects_list(context: Context) -> None:
```

Comment:
Didn't we call update project list when we call the workspaces? This could be a double operation, worth checking.